harness/determined/pytorch/_pytorch_context.py

```diff
@@ -800,6 +803,29 @@ def current_train_batch(self) -> int:
             raise det.errors.InternalException("Training hasn't started.")
         return self._current_batch_idx
 
+    def get_tensorboard_writer(self) -> Any:
+        """
+        Return the context's tensorboard writer to the user if the user needs it.
+
+        This method ensures we create a writer only when necessary
+        and imports SummaryWriter at runtime.
+
+        The return type is Any to avoid creating an environment dependency
+        on the tensorboard module, which is required to import SummaryWriter.
+        """
+
+        if self._tbd_writer is None:
+            from torch.utils.tensorboard import SummaryWriter
+
+            self._tbd_writer = SummaryWriter(self.get_tensorboard_path())  # type: ignore
+
+        return self._tbd_writer
+
+    def maybe_reset_tbd_writer(self) -> None:
```

Comment:
is this intentionally part of the user-facing API?  Or should it have an `_` prefix?